notebooks/concepts/imviz_compass_mpl.ipynb

```diff
@@ -0,0 +1,251 @@
+    "         'PC1_1': -1.14852e-05,\n",
+    "         'PC1_2': 7.01477e-06,\n",
+    "         'PC2_1': 7.75765e-06,\n",
+    "         'PC2_2': 1.20927e-05,\n",
+    "         'CDELT1': 1.0,\n",
+    "         'CDELT2': 1.0,\n",
+    "         'CUNIT1': 'deg',\n",
+    "         'CUNIT2': 'deg',\n",
+    "         'CTYPE1': 'RA---TAN',\n",
+    "         'CTYPE2': 'DEC--TAN',\n",
+    "         'CRVAL1': 3.581704851882,\n",
+    "         'CRVAL2': -30.39197867265,\n",
+    "         'LONPOLE': 180.0,\n",
+    "         'LATPOLE': -30.39197867265,\n",
+    "         'MJDREF': 0.0,\n",
+    "         'RADESYS': 'ICRS'})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7bd46c49",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wcs_utils.draw_compass_mpl(a, w)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c72b24e6-78ec-411d-aaaf-53babff6fea4",
+   "metadata": {},
+   "source": [
+    "## Case 4: GWCS\n",
+    "\n",
+    "Again, we use random data. The GWCS is from https://gwcs.readthedocs.io/en/latest/#getting-started example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1ff358a3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = np.random.random((1024, 2048))\n",
+    "\n",
+    "shift_by_crpix = models.Shift(-(2048 - 1) * u.pix) & models.Shift(-(1024 - 1) * u.pix)\n",
+    "matrix = np.array([[1.290551569736E-05, 5.9525007864732E-06],\n",
+    "                   [5.0226382102765E-06, -1.2644844123757E-05]])\n",
+    "rotation = models.AffineTransformation2D(matrix * u.deg, translation=[0, 0] * u.deg)\n",
+    "rotation.input_units_equivalencies = {\"x\": u.pixel_scale(1 * (u.deg / u.pix)),\n",
+    "                                      \"y\": u.pixel_scale(1 * (u.deg / u.pix))}\n",
+    "rotation.inverse = models.AffineTransformation2D(np.linalg.inv(matrix) * u.pix,\n",
+    "                                                 translation=[0, 0] * u.pix)\n",
+    "rotation.inverse.input_units_equivalencies = {\"x\": u.pixel_scale(1 * (u.pix / u.deg)),\n",
+    "                                              \"y\": u.pixel_scale(1 * (u.pix / u.deg))}\n",
+    "tan = models.Pix2Sky_TAN()\n",
+    "celestial_rotation = models.RotateNative2Celestial(\n",
+    "    5.63056810618 * u.deg, -72.05457184279 * u.deg, 180 * u.deg)\n",
+    "det2sky = shift_by_crpix | rotation | tan | celestial_rotation\n",
+    "det2sky.name = \"linear_transform\"\n",
+    "detector_frame = cf.Frame2D(name=\"detector\", axes_names=(\"x\", \"y\"), unit=(u.pix, u.pix))\n",
+    "sky_frame = cf.CelestialFrame(reference_frame=ICRS(), name='icrs', unit=(u.deg, u.deg))\n",
+    "pipeline = [(detector_frame, det2sky), (sky_frame, None)]\n",
+    "w = gwcs.wcs.WCS(pipeline)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e99e5d27",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wcs_utils.draw_compass_mpl(a, w)"
```

Comment:
Ginga preview not possible due to asdf-format/asdf#1032 , as the Ginga GUI requires the GWCS to be written to ASDF first so the underlying I/O can read it back in. In this notebook, we bypassed the bug by passing in GWCS directly without writing to ASDF first.

![dummy_gwcs_imviz](https://user-images.githubusercontent.com/2090236/144471080-bb5c9583-f442-487a-8ea9-d058dbbace6d.jpg)